src/Features/CSharp/Portable/CodeRefactorings/ConvertLocalFunctionToMethod/CSharpConvertLocalFunctionToMethodCodeRefactoringProvider.cs

```diff
@@ -42,15 +42,18 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
             var cancellationToken = context.CancellationToken;
             var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
 
-            var identifier = await root.SyntaxTree.GetTouchingTokenAsync(context.Span.Start,
-                token => token.Parent.IsKind(SyntaxKind.LocalFunctionStatement), cancellationToken).ConfigureAwait(false);
+            var identifier = context.Span.IsEmpty
```

Comment:
I am wondering should just the following check suffice instead of dealing with tokens/spans etc.?

```csharp
if (!(root.FindNode(context.Span) is LocalFunctionStatementSyntax localFunction))
{
   return;
}

// Else all the logic after localFunction is computed
```

Basically why do we need to care about `identifier` being touched or selected here at all?

Reply:
If we decide that for empty span scenario, we only want to offer the refactoring if we are explicitly on the local function identifier, then we can have an additional check after the code above:

```csharp
if (context.Span.IsEmpty && !localFunction.Identifier.Span.Contains(context.Span.Start)
{
   return;
}
```

Reply:
I might be wrong here but if `FindNode` finds the first node that matches the span that it wouldn't work for situation where only the identifier is selected. That has pretty easy solution, however. (could simply test `GetTouchingTokenAsync` even for selection and proceed with this test only for situations where it doesn't return anything valuable).

Reply:
> If we decide that for empty span scenario, we only want to offer the refactoring if we are explicitly on the local function identifier, then we can have an additional check after the code above:
> if (context.Span.IsEmpty && !localFunction.Identifier.Span.Contains(context.Span.Start)
> {
>    return;
> }

I don't quite follow. The `root.FindNode(context.Span)` would work only for situations where the whole local function is selected so we'd still need `GetTouchingTokenAsync` path for situations where carrot is within identifier / the identifier is selected. And for those we already know that `!localFunction.Identifier.Span.Contains(context.Span.Start)` is false. 

PS: I'm just trying to understand the general approach to things in the codebase that's why I'm asking all these questions.

Reply:
AFAIK The only alternative would be something like this (it's ugly quick prototype): 
```csharp
            var identifier = await root.SyntaxTree.GetTouchingTokenAsync(context.Span.Start, token => token.Parent.IsKind(SyntaxKind.LocalFunctionStatement), cancellationToken).ConfigureAwait(false);
            if (identifier == default && !context.Span.IsEmpty)
            {
                var tmpNode = root.FindNode(context.Span);
                if (tmpNode is LocalFunctionStatementSyntax tmpLf)
                {
                    identifier = tmpLf.Identifier;
                }
            }

            if (identifier == default)
            {
                return;
            }
```

Honestly can't say which way is better.

Reply:
FindNode does not require the passed in span to match the returned node’s span, it returns the smallest node whose FullSpan contains the input span. So, it will succeed if the input is empty span and span start position is on an immediate child token or selected span is an entire child token or it includes multiple child nodes and tokens.

Reply:
Ah, I see. I was looking at wrong FindNode. Ok, that would make things
substantially clearer.

Will change it tomorrow. Thanks for the input, starting to understand the
system better now :).

Reply:
It is still necessary to handle situation where Context.Span is empty due to `"C LocalFunction[||](C c)"` situation. It's feels cleaner to go with .FindNode for non-empty selection, however.